racket/collects/net/http-client.rkt

```diff
@@ -97,10 +103,15 @@
   ;; (thread (λ () (copy-port log-i to (current-error-port))))
 
   (set-http-conn-to! hc to)
-  (set-http-conn-from! hc from))
+  (set-http-conn-from! hc from)
+
+  (set-http-conn-auto-reconnect?! hc auto-reconnect?)
+  (set-http-conn-auto-reconnect-host! hc host-bs)
```

Comment:
Note, however, that `http-conn-close!` does **not** overwrite `port`, so I was able to reuse that field without having to add a new `auto-reconnect-port` field to `http-conn`.